upload/src/addons/SV/WarningImprovements/XF/Entity/Warning.php

```diff
@@ -13,6 +13,12 @@ class Warning extends XFCP_Warning
     public function canViewIssuer()
```

Comment:
Note; need to extend `canView()` as well to allow users to see their own warnings.

Reply:
Done

Reply:
I was expecting something like ```$this->user_id && $this->user_id == \XF::visitor()->user_id && \XF::options()->sv_view_own_warnings || parent::canView($error)``` (tho maybe formatted better), since there isn't any particular permission gating on it just the global option.

When viewing the warning, `canViewIssuer()` should then be used to see if they can access sensitive stuff